python/rmm/rmm/_lib/helper.pyx

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helper functions for rmm"""
+
+import re
+
+
+cdef dict BYTE_SIZES = {
+    'b': 1,
+    '': 1,
+    'kb': 1000,
+    'mb': 1000**2,
+    'gb': 1000**3,
+    'tb': 1000**4,
+    'pb': 1000**5,
+    'k': 1000,
+    'm': 1000**2,
+    'g': 1000**3,
+    't': 1000**4,
+    'p': 1000**5,
+    'kib': 1024,
+    'mib': 1024**2,
+    'gib': 1024**3,
+    'tib': 1024**4,
+    'pib': 1024**5,
+    'ki': 1024,
+    'mi': 1024**2,
+    'gi': 1024**3,
+    'ti': 1024**4,
+    'pi': 1024**5,
+}
+
+
+pattern = re.compile(r"^([0-9]+(?:\.[0-9]*)?)[\t ]*((?i:(?:[kmgtp]i?)?b)?)$")
```

Comment:
@Matt711 I don't see that the `b` is optional here?

Reply:
Ah I though I made it optional. What do you think, should it be optional or not? @wence-

Reply:
I've never seen `ki` as a prefix for `1024`, I'd be inclined to not support `1k` and `1ki` style strings (given the function is called `parse_bytes`, the units of bytes are `kb`, `kib` etc...).

Reply:
Yeah I've seen `k` and `ki`, particular in computer engineering applications. But `1k` vs `1kb` is not a big deal for me, so I'll apply your suggestions.